pydantic/main.py

```diff
@@ -629,53 +621,68 @@ def _iter(
         self,
         to_dict: bool = False,
         by_alias: bool = False,
-        allowed_keys: Optional['SetStr'] = None,
         include: Union['AbstractSetIntStr', 'DictIntStrAny'] = None,
         exclude: Union['AbstractSetIntStr', 'DictIntStrAny'] = None,
         exclude_unset: bool = False,
         exclude_defaults: bool = False,
         exclude_none: bool = False,
     ) -> 'TupleGenerator':
 
+        allowed_keys = self._calculate_keys(include=include, exclude=exclude, exclude_unset=exclude_unset)
+        if allowed_keys is None and not any((to_dict, by_alias, exclude_unset, exclude_defaults, exclude_none)):
+            # huge boost for plain _iter()
+            yield from self.__dict__.items()
+            return
+
         value_exclude = ValueItems(self, exclude) if exclude else None
         value_include = ValueItems(self, include) if include else None
 
+        iteritems: Union[ItemsView[str, Any], 'TupleGenerator'] = self.__dict__.items()
+        # adding only needed checks/conversions to generator
+        if allowed_keys is not None:
+            iteritems = ((k, v) for k, v in iteritems if k in allowed_keys)
+        if exclude_none:
+            iteritems = ((k, v) for k, v in iteritems if v is not None)
```

Comment:
I guess even if we have one loop we could move this check to before `_get_value`.

Reply:
Yeah, that surely good idea